project1/client/input_parser/input_parser.h

```diff
@@ -0,0 +1,105 @@
+#ifndef PROJECT1_INPUT_PARSER_H
+#define PROJECT1_INPUT_PARSER_H
+
+#include "../../wire_protocol/wire_protocol.h"
+#include "../../server/file_handler/my_file_handler.h"
+#include "ftp_messages.pb.h"
+#include <sstream>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+namespace client::input_parser {
+
+    /**
+     * @brief Assists in mapping input commands for FTP client to corresponding
+     * protobuf requests.
+     *
+     * @note There's definitely some redundancy here. Will likely change
```

Comment:
I'm not sure how much you can slim this down, considering that each request type requires somewhat unique code to populate it. If you really want to pretify this code, we can talk about it at one of our meetings; I have a few tricks up my sleeve for dealing with situations like this.